src/napari_spatialdata/_widgets.py

```diff
@@ -232,6 +240,65 @@ def _(self, vec: pd.Series, key: str, layer: Layer) -> dict[str, Any]:
         }
 
 
+class ScatterListWidget(AListWidget):
+    attrChanged = Signal()
+    _text = None
+
+    def __init__(self, viewer: Viewer, model: ImageModel, attr: str, **kwargs: Any):
+        AListWidget.__init__(self, viewer, model, attr, **kwargs)
+        self.attrChanged.connect(self._onChange)
+        self.data = -1
+
+    def _onAction(self, items: Iterable[str]) -> None:
+        for item in sorted(set(items)):
+            try:
+                vec, _ = self._getter(item, index=self.getIndex())
+            except Exception as e:  # noqa: B902
+                logger.error(e)
+                continue
+            self.data = vec
```

Comment:
I think the attribute `.data` should be either set as a property for proper getter/setter or should not be used and the vec should just be returned

Reply:
because of the way onAction is connected to different events I think it would be easier to set data as property (I think it would be hard to handle a return in some cases) so I will do that then if thats ok